account_invoice_date_due/models/account_move.py

```diff
@@ -0,0 +1,60 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # Show invoice date due even when payment term is defined
+    invoice_date_due_payment_term = fields.Date(
+        related="invoice_date_due", string="Due Date Payment Term"
+    )
+    invoice_date_due = fields.Date(
+        string="Due Date",
+        compute="_compute_invoice_date_due",
+        inverse="_inverse_invoice_date_due",
+    )
+
+    def _inverse_invoice_date_due(self):
+        for invoice in self:
+            if invoice.state == "posted":
+                if not self.env.user.has_group(
+                    "account_invoice_date_due.allow_to_change_due_date"
+                ):
+                    raise UserError(_("You are not allowed to change the due date."))
+
+    @api.onchange("invoice_date_due_payment_term")
+    def _onchange_invoice_date_due_payment_term(self):
+        """Propagate from Payment term due date to original field"""
+        if self.invoice_date_due_payment_term:
+            self.invoice_date_due = self.invoice_date_due_payment_term
+
+    def _compute_amount(self):
+        # Avoid recomputation of amount fields when we are only changing the
+        # invoice's due date
+        if self.env.context.get("bypass_compute_amount"):
```

Comment:
I tried to write a unit test for better coverage in order to check if super  `_compute_amount` is called when invoice's due date is updated.
I did some functional testing and indeed the method is not called. 
When I look at the  dependencies of ` _compute_amount` I don't see any related with `invoice_date_due`  https://github.com/odoo/odoo/blob/a6f8d9c512ffb816e18bb43a5efa510707877cf6/addons/account/models/account_move.py#L833
The variable `bypass_compute_amount` seems to not be required